pkg/kubectl/cmd/util/factory.go

```diff
@@ -355,9 +405,9 @@ func (f *factory) DiscoveryClient() (discovery.CachedDiscoveryInterface, error)
 	return NewCachedDiscoveryClient(discoveryClient, cacheDir, time.Duration(10*time.Minute)), nil
 }
 
-func (f *factory) Object() (meta.RESTMapper, runtime.ObjectTyper) {
+func Object(dcf DiscoveryClientFactory, ccf ClientCacheFactory) (meta.RESTMapper, runtime.ObjectTyper) {
```

Comment:
Without polymorphism, this just promotes bad habits and inscrutable code as evidenced by what we had before. If we need rules about genericism for anything being referneced (more likely), I'd rather enforce those.